phys/module_physics_init.F

```diff
@@ -5198,4 +5216,30 @@ subroutine interp_vec_cu(locvec,locwant,periodic,loc1,loc2,wght1,wght2)
   end subroutine interp_vec_cu
 !---------------------END   PSH/TWG  CHANGES----------------
 
+  subroutine compute_2d_dx_area(dx, dy, msftx, msfty, dx2d, area2d, &
+                                ids, ide, jds, jde, kds, kde,       &
+                                ims, ime, jms, jme, kms, kme,       &
+                                its, ite, jts, jte, kts, kte)
+
+     implicit none
+
+     integer , intent(in) :: ids, ide, jds, jde, kds, kde,       &
+                             ims, ime, jms, jme, kms, kme,       &
+                             its, ite, jts, jte, kts, kte
+
+     real,                             intent(in)  :: dx, dy
+     real, dimension(ims:ime,jms:jme), intent(in)  :: msftx, msfty
+     real, dimension(ims:ime,jms:jme), intent(out) :: dx2d, area2d
+
+     integer :: i, j
+
+     do j = jts, min(jde-1,jte)
+        do i = its, min(ide-1,ite)
+           dx2d  (i,j) = sqrt ( dx/msftx(i,j) * dy/msfty(i,j) )
+           area2d(i,j) =        dx/msftx(i,j) * dy/msfty(i,j)
```

Comment:
Should we just change these two lines to be:
```
           area2d(i,j) =        dx/msftx(i,j) * dy/msfty(i,j)
           dx2d  (i,j) = sqrt ( area2d(i,j) )
```

Reply:
@mgduda 
Michael,
Agreed and done
first: compute grid cell area
second: compute dx = sqrt(area)